### ISSUE TITLE
Snackbar is hidden by iOS keyboard. So using it for validation is tricky.

### ISSUE THREAD

**Issue by firescript** [bradmartin/nativescript-snackbar#5]
If we could set a position property, to maybe say bottom, or top. That would be useful. Just in cases where developers are using it to handle validation alerts.

**Comment by bradmartin** [comment]
I don't have much iOS knowledge so I'm not sure how we could handle this. Especially since we are using a cocoapod for the iOS version.

@sitefinitysteve - do you know of a way to handle this scenario?

**Comment by firescript** [comment]
I asked here as well: https://github.com/stonesam92/SSSnackbar/issues/6

**Comment by sitefinitysteve** [comment]
@bradmartin Howbdies android handle the scenario... Or does it

**Comment by bradmartin** [comment]
It doesn't from what I recall. I will check later.

**Comment by bradmartin** [comment]
It's not really the purpose of a Snackbar to use for validation. On a form
you would use it for informing the user the form didn't submit
successfully. With the option to resubmit. So the keyboard typically isn't
open, but on android I believe the default works well with the keyboard
being open. So if the cocoapod could just handle the keyboard the same,
that would be good.

**Comment by firescript** [comment]
On android does it just slide up above the keyboard?

**Comment by firescript** [comment]
I just put in a PR that has support for show and showTop methods now. show() uses the bottom default way, and showTop() renders it at the top of the view.

https://github.com/stonesam92/SSSnackbar/pull/7

**Comment by bradmartin** [comment]
Great work.

**Comment by sitefinitysteve** [comment]
I think for this {N} plugin we should keep the "show" but pass in an optional position parameter... since it has to work on both platforms, internally android could just ignore.

**Comment by bradmartin** [comment]
Sounds good. Once they update the cocoapod, can one of you guys handle the
update? I don't have a Mac to do test with (I know, I know... very sad).

**Comment by sitefinitysteve** [comment]
yepyep

**Comment by sitefinitysteve** [comment]
Although looks like android has a variant too? http://stackoverflow.com/questions/30585550/how-do-i-change-an-android-snackbars-initial-alignment-from-bottom-to-top

**Comment by bradmartin** [comment]
Yea I don't care to implement this, as it goes against the material design
spec on what a Snackbar is. If you guys want to do it then that's fine as
long as its an optional feature. It's just not an issue on android, if the
cocoapod managed the layout according to the spec then putting it at the
top wouldn't be an issue.

At any rate if you guys want it and have the time to implement it then
thats great, I don't have the time right now and this isn't something I'm
in favor or. However I respect you guys and if you want/need it then that
gains my attention to make the plugin better :)

**Comment by bradmartin** [comment]
Any update on the iOS stuff?

**Comment by sitefinitysteve** [comment]
Nope, looks like that pod is just orphaned atm :/  Leave this issue open though if you could

**Comment by Obsessive** [comment]
I think we can use this line of code to dismiss the keyboard without worrying about which view is the first responder.

In Swift
UIApplication.sharedApplication().sendAction("resignFirstResponder", to:nil, from:nil, forEvent:nil)

In Objective-C
[[UIApplication sharedApplication] sendAction:@selector(resignFirstResponder) to:nil from:nil forEvent:nil];

**Comment by Obsessive** [comment]
So before showing snackbar we can hide the keyboard

**Comment by bradmartin** [comment]
If one of the other iOS guys want to test and confirm this approach let's
add it :)
